cmake/TPLs/FindTPLgtest.cmake

```diff
@@ -1,5 +1,26 @@
+set(REQUIRED_HEADERS gtest.h)
+set(REQUIRED_LIBS_NAMES "gtest googletest")
+set(IMPORTED_TARGETS_FOR_ALL_LIBS GTest::gtest)
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( gtest
-  REQUIRED_HEADERS gtest/gtest.h 
-  REQUIRED_LIBS_NAMES gtest
+tribits_tpl_allow_pre_find_package(gtest GTEST_ALLOW_PREFIND)
+
+if (GTEST_ALLOW_PREFIND)
+  message("-- Using find_package(GTest ...) ...")
+  find_package(GTest)
+  if (GTest_FOUND)
+    message("-- Found GTest_DIR='${GTest_DIR}'")
+    message("-- Generating gtest::all_libs and gtestConfig.cmake")
+    tribits_extpkg_create_imported_all_libs_target_and_config_file(gtest
+      INNER_FIND_PACKAGE_NAME GTest
+      IMPORTED_TARGETS_FOR_ALL_LIBS  ${IMPORTED_TARGETS_FOR_ALL_LIBS})
```

Comment:
> The cmake variables Trilinos_ENABLE_Gtest and TPL_ENABLE_Gtest need that spelling (i.e., lower-case t after G) but my installation of googletest sets up the cmake file as GTestConfig.cmake (notice the capital T).
>
> I didn't figure out a workable solution yet. Has anyone else noticed this or dealt with it?

@alanw0, as shown here, the TriBITS name of the TPL/external package is `gtest` while the name of the package-config file will be `CTestkConfig.cmake`.  Therefore, you enable this TPL with `-D TPL_ENABLE_gtest=ON`.